### PR TITLE
fix: labels in enrollment selection

### PR DESCRIPTION
**Pull request by anipendakur-okta**
OKTA-313164

## PR Checklist

- [X] Have you verified the basic functionality for this change?

- [X] Added unit tests?

- [ ] Added e2e tests

- [X] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Screenshot/Video:
image

### Reviewers:
@haishengwu-okta @magizh-okta 

### Issue:

- [OKTA-313164](https://oktainc.atlassian.net/browse/OKTA-313164)

### CODE REVIEW

File: packages/@okta/i18n/src/properties/login.properties [hunk 1]
```diff
@@ -699,15 +699,15 @@ cert.authentication.title = Certificate authentication
 ##################################################
 
 ## Password
-oie.password = Password
+oie.password.authenticator.description = Choose a password for your account
```
Comment (anipendakur-okta) [inline packages/@okta/i18n/src/properties/login.properties:702]: The changes here is to make the labels more specific in compliance with i18nTransformer labels.

File: src/v2/ion/i18nTransformer.js [hunk 2]
```diff
@@ -45,11 +45,17 @@ const I18N_OVERRIDE_MAPPINGS = {
   'identify.credentials.passcode': 'primaryauth.password.placeholder',
   'identify.rememberMe': 'remember',
 
-  'select-authenticator-enroll.authenticator.email': 'oie.email',
-  'select-authenticator-enroll.authenticator.password': 'oie.password',
-  'select-authenticator-enroll.authenticator.phone': 'oie.phone',
+  'select-authenticator-enroll.authenticator.email': 'oie.authenticator.email.label',
```
Comment (anipendakur-okta) [inline src/v2/ion/i18nTransformer.js:48]: Made these labels more specific for future. Also removed labels to accommodate what comes from response

File: src/v2/ion/i18nTransformer.js [hunk 3]
```diff
@@ -88,6 +88,7 @@ const getI18nKey = (i18nPath) => {
 
   if (i18nKey && !Bundles.login[i18nKey]) {
     Logger.warn(`expect i18n key ${i18nKey} for ${i18nPath} but not found in 'login' bundle.`);
+    return false;
```
Comment (anipendakur-okta) [inline src/v2/ion/i18nTransformer.js:91]: This is needed for line #106 in this file.

File: src/v2/view-builder/components/AuthenticatorEnrollOptions.js [hunk 4]
```diff
@@ -22,13 +22,13 @@ const AuthenticatorRow = View.extend({
         </div>\
         <div class="authenticator-description">\
           <h3 class="authenticator-label">{{label}}</h3>\
-          {{#if factorDescription}}\
-            <p>{{factorDescription}} </p>\
+          {{#if description}}\
```
Comment (anipendakur-okta) [inline src/v2/view-builder/components/AuthenticatorEnrollOptions.js:25]: Corrected the key.

File: src/v2/view-builder/utils/FactorUtil.js [hunk 5]
```diff
@@ -14,26 +14,22 @@ import { getPasswordComplexityDescriptionForHtmlList } from '../../../util/Facto
 
 const factorData = {
   'email': {
-    label: loc('oie.email', 'login'),
```
Comment (anipendakur-okta) [inline src/v2/view-builder/utils/FactorUtil.js:17]: Removing labels from here in order to ensure we accept what comes from backend.

File: packages/@okta/i18n/src/properties/login.properties (hunk 1 above)
Comment (magizh-okta) [inline packages/@okta/i18n/src/properties/login.properties:702]: Same is used in verify ?

Comment (anipendakur-okta) [reply packages/@okta/i18n/src/properties/login.properties:702]: There are no descriptions in verify... So it's only for enroll step

File: src/v2/ion/i18nTransformer.js [hunk 6]
```diff
@@ -45,11 +45,17 @@ const I18N_OVERRIDE_MAPPINGS = {
   'identify.credentials.passcode': 'primaryauth.password.placeholder',
   'identify.rememberMe': 'remember',
 
-  'select-authenticator-enroll.authenticator.email': 'oie.email',
-  'select-authenticator-enroll.authenticator.password': 'oie.password',
-  'select-authenticator-enroll.authenticator.phone': 'oie.phone',
+  'select-authenticator-enroll.authenticator.email': 'oie.authenticator.email.label',
+  'select-authenticator-enroll.authenticator.password': 'oie.authenticator.password.label',
+  'select-authenticator-enroll.authenticator.phone': 'oie.authenticator.phone.label',
   'select-authenticator-enroll.authenticator.security_key': 'oie.webauthn',
```
Comment (haishengwu-okta) [inline src/v2/ion/i18nTransformer.js:51]: change format for keys `oie.webauthn` and `oie.app` as well.

File: test/unit/spec/v2/ion/i18nTransformer_spec.js [hunk 7]
```diff
@@ -877,5 +877,50 @@ describe('v2/ion/i18nTransformer', function () {
...
+          relatesTo: {
+            value: {
+              type: 'password'
+            }
+          },
+          name: 'challenge-authenticator',
+          uiSchema: [
+            {
+              'name': 'non.existent.key',
+              'label': 'Password',
```
Comment (haishengwu-okta) [inline test/unit/spec/v2/ion/i18nTransformer_spec.js:893]: make the label a bit special so we know it is from unit test.

File: src/v2/ion/i18nTransformer.js (hunk 6 above)
Comment (magizh-okta) [reply src/v2/ion/i18nTransformer.js:51]: @anipendakur-okta  can you please update the oie.app to oie.okta.verify please. Thanks! I dont want to create a new pr for this.

File: src/v2/view-builder/utils/FactorUtil.js [hunk 8]
```diff
@@ -45,14 +41,13 @@ const factorData = {
   },
```
Comment (anipendakur-okta) [inline src/v2/view-builder/utils/FactorUtil.js:41]: @magizh-okta line #38 - I have not removed it like other authenticators, so the label is hard-coded to come from the props file. Want me to change?

File: src/v2/view-builder/utils/FactorUtil.js [hunk 9]
```diff
@@ -45,14 +41,13 @@ const factorData = {
...
 
   'security_key': {
-    label: loc('oie.webauthn', 'login'),
     description: loc('oie.webauthn.description', 'login'),
     iconClassName: 'mfa-webauthn',
   },
 
   'app': {
     label: loc('oie.app', 'login'),
```
Comment (anipendakur-okta) [inline src/v2/view-builder/utils/FactorUtil.js:49]: @magizh-okta line #49 - I have not removed it like other authenticators, so the label is hard-coded to come from the props file. Want me to change? ( I will change the key but I may have to remove it here if we're taking the same route as other authenticators )

File: src/v2/view-builder/utils/FactorUtil.js [hunk 10]
```diff
@@ -14,45 +14,39 @@ import { getPasswordComplexityDescriptionForHtmlList } from '../../../util/Facto
...
   'security_key': {
-    label: loc('oie.webauthn', 'login'),
     description: loc('oie.webauthn.description', 'login'),
     iconClassName: 'mfa-webauthn',
   },
 
   'app': {
-    label: loc('oie.app', 'login'),
-    description: '',
+    label: loc('oie.authenticator.okta_verify.label', 'login'),
```
Comment (haishengwu-okta) [inline src/v2/view-builder/utils/FactorUtil.js:48]: Update the mapping in i18nTransformer then the `label` can be removed.

File: src/v2/view-builder/utils/FactorUtil.js (hunk 8 above)
Comment (magizh-okta) [reply src/v2/view-builder/utils/FactorUtil.js:40]: we want to override it can you please put back the label here.

Comment (magizh-okta) [reply src/v2/view-builder/utils/FactorUtil.js:40]: I guess Haisheng meant to leave a comment to update the label key as oie.authenticator.webauthn.label'

File: src/v2/view-builder/utils/FactorUtil.js (hunk 10 above)
Comment (anipendakur-okta) [reply src/v2/view-builder/utils/FactorUtil.js]: Spoke with @magizh-okta -
* Okta Verify label comes from backend.
    - Removed the entry (`oie.app`) from `login.properties`.
    - Added a key (`oie.authenticator.app.label`) in `i18nTransformer` like other authenticators.

* Webauthn label comes from `login.properties`. Updated login.properties with a key (`oie.authenticator.webauthn.label`) and updated `i18nTransformer`.

File: src/v2/view-builder/utils/FactorUtil.js (hunk 8 above)
Comment (anipendakur-okta) [reply src/v2/view-builder/utils/FactorUtil.js:40]: Updated according to our call.

**Pull request merged**